src/conq/data_recorder.py

```diff
@@ -94,14 +93,17 @@ def save_episode_worker(self, robot_state_client, done: Event, mode: str):
 
         while not done.is_set():
             now = time.time()
-            state = robot_state_client.get_robot_state()
+            state = clients.state.get_robot_state()
+            localization_state = clients.graphnav.get_localization_state()
 
             from conq.rerun_utils import viz_common_frames
             viz_common_frames(state.kinematic_state.transforms_snapshot)
 
             step_data = {
                 'time': now,
                 'robot_state': state,
+                'localization': localization_state.localization,
+                'is_lost': localization_state.lost_detector_state.is_lost,
```

Comment:
Has this been tested when the GraphNav localization service wasn't running? In general, do these client requests error out when there's no client set (e.g. we don't want to record images so we don't set the image client) or when the corresponding service isn't running on the robot (localization in this case)

Reply:
I should clarify that this is a genuine question and not a blocker for merging this PR, at least for me personally. Might be worth striking up a quick convo with Peter if you all are both in the lab tomorrow

Reply:
Good call! We would not want to call a function on a `None` type object which would break the recorder. Modified in ec9c2588a434ca88d7f2016a982040f3b3dc2a40 @PeterMitrano please double check these changes make sense